api/apiclient.go

```diff
@@ -658,9 +711,37 @@ func dialWebsocketMulti(ctx context.Context, addrs []string, path string, opts d
 	return result.(*dialResult), nil
 }
 
+func lookupIPAddr(ctx context.Context, host string, resolver IPAddrResolver) ([]string, error) {
+	addrs, err := resolver.LookupIPAddr(ctx, host)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	ips := make([]string, 0, len(addrs))
+	for _, addr := range addrs {
+		if addr.Zone != "" {
+			// Ignore IPv6 zone. Hopefully this shouldn't
+			// cause any problems in practice.
+			logger.Infof("ignoring IP address with zone %q", addr)
+			continue
+		}
+		ips = append(ips, addr.IP.String())
```

Comment:
Currently the zoned IP addresses aren't parsable by any of our network code - they're not understood as valid IP addresses AFAIK, so I think it's best to exclude them unless there's a good reason not to.